src/main/scala/nl/knaw/dans/easy/multideposit/actions/AddFileMetadataToDeposit.scala

```diff
@@ -18,77 +18,103 @@ package nl.knaw.dans.easy.multideposit.actions
+    }
+    else // this means the dataset does not contain any data
+      Try { List.empty }
+  }.flatten
+
   /**
    * Verifies whether all preconditions are met for this specific action.
    * All files referenced in the instructions are checked for existence.
    *
    * @return `Success` when all preconditions are met, `Failure` otherwise
    */
   override def checkPreconditions: Try[Unit] = {
-    val inputDir = settings.multidepositDir
-    // Note that the FILE_SIP paths are not really used in this action
-    val nonExistingFileSIPs = dataset.getOrElse("FILE_SIP", List.empty)
-      .filterNot(_.isEmpty)
-      .filterNot(fp => new File(settings.multidepositDir, fp).exists())
-
-    if (nonExistingFileSIPs.isEmpty) Success(())
-    else Failure(ActionException(row, s"""The following SIP files are referenced in the instructions but not found in the deposit input dir "$inputDir" for dataset "$datasetID": ${ nonExistingFileSIPs.mkString("[", ", ", "]") }""".stripMargin))
-  }
+    import validators._
 
-  override def execute(): Try[Unit] = writeFileMetadataXml(row, datasetID)
-}
-object AddFileMetadataToDeposit {
+    def checkSFColumnsIfDatasetContainsAVFiles(mimetypes: List[(File, String)]): Try[Unit] = {
+      val avFiles = mimetypes.filter {
+        case (_, mimetype) => (mimetype startsWith "video") || (mimetype startsWith "audio")
+      }
+
+      if (avFiles.nonEmpty)
+        checkColumnsAreNonEmpty(row, dataset, "SF_USER", "SF_COLLECTION")
+          .recoverWith {
+            case ActionException(r, msg, _) => Failure(ActionException(r,
+              s"$msg\ncause: these columns should contain values because audio/video files are " +
+                s"found:\n${ avFiles.map { case (file, _) => s" - $file" }.mkString("\n") }"))
+          }
+      else
+        checkColumnsAreEmpty(row, dataset, "SF_DOMAIN", "SF_USER", "SF_COLLECTION")
+          .recoverWith {
+            case ActionException(r, msg, _) => Failure(ActionException(r,
+              s"$msg\ncause: these columns should be empty because there are no audio/video " +
+                s"files found in this dataset"))
+          }
+    }
+
+    mimetypeMap.flatMap(checkSFColumnsIfDatasetContainsAVFiles)
+  }
 
-  def writeFileMetadataXml(row: Int, datasetID: DatasetID)(implicit settings: Settings): Try[Unit] = {
-    datasetToFileXml(datasetID)
+  override def execute(): Try[Unit] = {
+    datasetToFileXml
       .map(stagingFileMetadataFile(datasetID).writeXml(_))
       .recoverWith {
         case NonFatal(e) => Failure(ActionException(row, s"Could not write file meta data: $e", e))
       }
   }
 
-  def datasetToFileXml(datasetID: DatasetID)(implicit settings: Settings): Try[Elem] = {
-    val inputDir = multiDepositDir(datasetID)
+  private def datasetToFileXml: Try[Elem] = {
+    mimetypeMap.map(fileXmls(_) match {
+      case Nil => <files xmlns:dcterms="http://purl.org/dc/terms/"/>
+      case files =>
+        // @formatter:off
```

Comment:
Can't we configure the behaviour of the formatter? On the other hand, if some developer hasn't the desired option installed it still might change unintentionally sooner or later.